python/pyarb.hpp

```diff
@@ -0,0 +1,84 @@
+#pragma once
```

Comment:
How about putting this content in `python/cable_probes.hpp`, because this is really specific to probes/samplers.

Reply:
It's probes and samplers, but not cable probes specifically.

Part of the motivation for this design is to keep the simulator code uncoupled from the cable probe definitions and wrappers. This way, the cable probe wrapper code is responsible for registering its metadata handlers, and the simulator code is responsible for using those handlers to process samples.